exe.c

```diff
@@ -272,6 +272,8 @@ int cgc_locals_fun = 0;
 int cgc_globals = 0;
 // Bump allocator used to allocate static objects
 int cgc_global_alloc = 0;
+// If the main function returns a value
+bool main_returns = false;
```

Comment:
Working on the tests revealed a few bugs in pnut-exe and pnut-sh.
- In pnut-exe, if the main function didn't return an exit code, the exit could would be whatever value that was left in the `reg_X` register.
- In pnut-exe, global variables live on the stack and were left unitialized. While on Linux, that seemed to be fine because the stack was all 0, on MacOS the stack contained values from the program loader.